app/views/PillarAnalysis/AnalyticalStatementInput/StoryAnalysisModal/index.tsx

```diff
@@ -315,6 +341,106 @@ function StoryAnalysisModal(props: Props) {
         message: 'Looks like there are some changes that have not been saved yet. Are you sure you want to close?',
     });
 
+    const variables = useMemo(() => (
+        isDefined(automaticSummaryId)
+            ? {
+                projectId,
+                summaryId: automaticSummaryId,
+            }
+            : undefined
+    ), [projectId, automaticSummaryId]);
+
+    const {
+        data,
+        loading,
+        startPolling,
+        stopPolling,
+        error,
+    } = useQuery<AnalyticalInformationSummaryQuery, AnalyticalInformationSummaryQueryVariables>(
+        ANALYTICAL_INFORMATION_SUMMARY,
+        {
+            skip: !automaticSummaryId,
+            variables,
+        },
+    );
+
+    const informationGapOnClick = useCallback(() => {
+        if (pristine) {
+            setPristine(false);
+        }
+        const informationGapResponse = data?.project?.analysisAutomaticSummary?.informationGap;
+        setInformationGaps(informationGapResponse);
+    }, [pristine, data?.project?.analysisAutomaticSummary]);
+
+    const analyticalStatementOnClick = useCallback(() => {
```

Comment:
```suggestion
    const onStatementGenerateClick = useCallback(() => {
```
Use better naming for all the functions following this one.